crates/autopilot/src/infra/persistence/mod.rs

```diff
@@ -140,6 +147,37 @@ impl Persistence {
 
         ex.commit().await.context("commit")
     }
+
+    /// Get normalized token prices.
+    pub async fn normalized_auction_prices(
+        &self,
+        auction: domain::auction::Id,
+    ) -> Result<HashMap<eth::TokenAddress, domain::auction::NormalizedPrice>, Error> {
+        let mut ex = self
+            .postgres
+            .pool
+            .begin()
+            .await
+            .context("begin")
+            .map_err(Error::DbError)?;
+
+        let unit = BigInt::from(1_000_000_000_000_000_000_u128);
+        let prices = database::auction_prices::fetch(&mut ex, auction)
+            .await
+            .context("fetch")
+            .map_err(Error::DbError)?
+            .into_iter()
+            .map(|p| {
+                (eth::H160(p.token.0).into(), {
+                    let price = big_decimal_to_u256(&p.price).unwrap();
+                    (price.to_big_rational() / &unit).into()
+                })
+            })
+            .collect();
+
+        let _ = ex.commit().await.context("commit");
```

Comment:
No need to commit the tx if we only `fetch` data from the DB and don't write anything.

Reply:
fixed